.github/workflows/upgrade-check.yml.sample

```diff
@@ -0,0 +1,67 @@
+# This workflow compares a downstream ansible-slurm-appliance repository for a specific site with the upstream
+# stackhpc/ansible-slurm-appliance repository to check whether there is a new upstream version available. If a
+# newer tag is found in the upstream repository then a pull request is created to the downstream repo
+# in order to merge in the changes from the new upstream release.
+
+# To use this workflow in a downstream ansible-slurm-appliance repository simply copy it into .github/workflows
+# and give it an appropriate name, e.g.
+# cp .github/workflows/upgrade-check.yml.sample .github/workflows/upgrade-check.yml
+
+name: Check for upstream updates
+on:
+  schedule:
+    - cron: "0 9 * * *"
+  workflow_dispatch:
+jobs:
+  check_for_update:
+    runs-on: ubuntu-22.04
+    # permissions:
+    #   contents: write
+    #   pull-requests: write
+    #   actions: write
+
+    steps:
+      - name: Checkout the config repo
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          fetch-tags: true
+
+      # Based on equivalent GitLab CI job
```

Comment:
Let's change this to say 'based on equivalent azimuth-config job' to make it clear what we based this workflow on. Might also be worth adding a link to that azimuth-config sample workflow.